### PR TITLE
Droplet: option to open documentation in new window

### PR DESCRIPTION
**Pull request by JillianK**
This adds a button in the documentation modal that opens the documentation in a new tab. Note that what opens will be whatever they originally clicked 'See examples' on, we don't track what they click on within the modal.
Sep-17-2020 09-10-09

## Links

- [spec](https://codedotorg.atlassian.net/browse/STAR-1242?atlOrigin=eyJpIjoiZjYzOWNlMmI3MDkzNGIyNzkxMjI0YzlmNmVjMDVkMmEiLCJwIjoiaiJ9)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by breville** [comment]
cc @Hamms who is our expert on implications of opening in a new tab.

**Comment by Hamms** [comment]
I'm _extremely_ hesitant to implement this functionality in a div with an event handler. There are a handful of reasons for this, accessibility issues being perhaps the most significant, but see the "best practices" section of https://developer.mozilla.org/en-US/docs/Web/API/Window/open for some more background on this specific solution.

More generally, this _does_ actually seem like one of those rare cases in which it is actually acceptable to open a link in a new tab. Specifically, we expect this to be in an interface in which the user has the possibility to lose work if they are unexpectedly navigated away, _and_ we're linking to an internal url so the potential security issues that are always inherent to any "open in a new tab" functionality are minimized.

I'd recommend implementing this with the `target=_blank` functionality of anchor tags, making absolutely sure to set `noopener` and `noreferrer` appropriately.

For more information, see:
- https://blog.dareboost.com/en/2017/03/target-blank-links-rel-noopener-performance-security/
- https://medium.com/sedeo/how-to-fix-target-blank-a-security-and-performance-issue-in-web-pages-2118eba1ce2f
- https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/

**Comment by cforkish** [comment]
@Hamms did you try playing around with the linter rule from the medium article?

**Comment by Hamms** [comment]
I haven't but I'd love it if we could have such a rule. The major hurdle we'd have to overcome is the sheer variety of ways in which such a link can be defined throughout our codebase.

File: apps/src/code-studio/LegacyDialog.js [hunk 1]
```diff
@@ -81,10 +81,28 @@ var LegacyDialog = (module.exports = function(options) {
...
     modalBody.append(closeLink);
+    if (options.link) {
+      modalBody.append(
+        $(
+          `<div><a target="_blank" href="${
+            options.link
+          }" rel="noopener noreferrer"></a></div>`
+        ).addClass('open-link')
+      );
+    }
```
Comment (maddiedierker) [inline apps/src/code-studio/LegacyDialog.js]: nit: move this functionality into a helper function

**Review by maddiedierker — APPROVED** [review]
left a nit to DRY up appending the link, but it looks like you've addressed elijah's concern and this LGTM!

**Pull request merged**